hi-ml/src/health_ml/run_ml.py

```diff
@@ -274,81 +268,78 @@ def is_crossval_disabled_or_child_0(self) -> bool:
+
+    def init_inference(self) -> None:
+        """ Prepare the runner for inference: validation or test. The following steps are performed:
+        1. Get the checkpoint to use for inference. This is either the checkpoint from the last training epoch or the
+        one specified in src_checkpoint argument.
+        2. If the container has a run_extra_val_epoch method, call it to run an extra validation epoch.
+        3. Create a new trainer instance for inference. This is necessary because the trainer is created with a single
+        device in contrast to training that uses DDP if multiple GPUs are available.
+        4. Create a new data module instance for inference to account for any requested changes in the dataloading
+        parameters (e.g. batch_size, max_num_workers, etc) as part of on_run_extra_validation_epoch.
+        """
+        self.inference_ckpt = str(self.checkpoint_handler.get_checkpoint_to_test())
+        if self.container.run_extra_val_epoch:
+            self.container.on_run_extra_validation_epoch()
+        self.set_trainer_for_inference()
+        self.data_module = self.container.get_data_module()
 
     def run_training(self) -> None:
         """
         The main training loop. It creates the Pytorch model based on the configuration options passed in,
         creates a Pytorch Lightning trainer, and trains the model.
         If a checkpoint was specified, then it loads the checkpoint before resuming training.
+        The cwd is changed to the outputs folder so that the model can write to current working directory, and still
+        everything is put into the right place in AzureML (only the contents of the "outputs" folder is treated as a
+        result file).
         """
-        # Change to the outputs folder so that the model can write to current working directory, and still everything
-        # is put into the right place in AzureML (only the contents of the "outputs" folder is treated as a result file)
         with change_working_directory(self.container.outputs_folder):
             assert self.trainer, "Trainer should be initialized before training. Call self.init_training() first."
-            self.trainer.fit(self.container.model, datamodule=self.container.get_data_module())
-
+            self.trainer.fit(self.container.model, datamodule=self.data_module)
         for logger in self.trainer.loggers:
             assert logger is not None
             logger.finalize('success')
 
     def run_validation(self) -> None:
-        """
-        Run validation on the validation set for all models to save time/memory consuming outputs.
-        """
-        self.container.on_run_extra_validation_epoch()
-        trainer = self.get_trainer_for_inference(checkpoint_path=None)
-        with change_working_directory(self.container.outputs_folder):
-            trainer.validate(self.container.model, datamodule=self.container.get_data_module())
-
-    def validate_model_weights(self) -> None:
```

Comment:
no need to validate this manually, rely on PL checkpoints weights loading. another advatage here is that we shouldn't worry about container/model states anymore cause PL will only load weights. When we load chcekpoint ourselve we should be careful to do it before calling on_run_extra_val_epoch otherwise model's state gets reset from the checkpoint